### PR TITLE
Added content app in the example and basic discovery and communication.

### PR DESCRIPTION
**Pull request by amitnj**
[DRAFT PR] to start review.

#### Problem
What is being fixed?  Examples:
Added content app in the tv-app example and basic discovery and communication.

#### Change overview
added tv-app -> content-app : example content app
added tv-app -> common-api : common api between tv platform-app and content-app
moved app -> platform-app : refactored earlier tv-app into platform-app with package name changes.

#### Testing
How was this tested? (at least one bullet point required)
* If unit tests were added, how do they cover this issue?
* If unit tests existed, how were they fixed/modified to prevent this in future?
* If new unit tests are not added, why not?
* If integration tests were added, how do they verify this change?
* If new integration tests are not added, why not?
* If manually tested, what platforms controller and device platforms were manually tested, and how?
* If no testing is required, why not?

### CODE REVIEW

**Review by woody-apple — APPROVED** [review]

File: examples/tv-app/android/App/common-api/build.gradle [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+    compileOptions {
+        sourceCompatibility JavaVersion.VERSION_1_8
+        targetCompatibility JavaVersion.VERSION_1_8
+    }
+//    publishing {
+//        multipleVariants {
+//            allVariants()
+//            withJavadocs()
+//        }
+//    }
```
Comment (lazarkov) [inline examples/tv-app/android/App/common-api/build.gradle:33]: Should be remover or uncommented?

File: examples/tv-app/android/App/common-api/build.gradle [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+//            }
+//        }
+//        repositories {
+//            maven {
+//                name = 'myRepo'
+//                url = mavenLocal().getUrl()
+//            }
+//        }
+//    }
+//}
```
Comment (lazarkov) [inline examples/tv-app/android/App/common-api/build.gradle:61]: Should be removed or uncommented?

File: examples/tv-app/android/App/content-app/build.gradle [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+
+}
+
+dependencies {
+
+    implementation 'androidx.appcompat:appcompat:1.3.1'
+    implementation 'com.google.android.material:material:1.4.0'
+    implementation 'androidx.constraintlayout:constraintlayout:2.1.3'
+    implementation project(path: ':common-api')
+    // implementation 'com.matter.tv.app.api:'
```
Comment (lazarkov) [inline examples/tv-app/android/App/content-app/build.gradle:37]: Should be uncommented?

File: examples/tv-app/android/App/platform-app/src/main/java/com/matter/tv/server/MainActivity.java [hunk 4]
```diff
@@ -0,0 +1,178 @@
...
+            }
+          }
+        };
+    registerReceiver(broadcastReceiver, new IntentFilter("com.matter.tv.server.appagent.add"));
+    registerReceiver(broadcastReceiver, new IntentFilter("com.matter.tv.server.appagent.remove"));
+
+    ContentAppDiscoveryService.getRecieverInstance().registerSelf(this.getApplicationContext());
+    //
+    // MatterContentAppDiscoveryService.getRecieverInstance().registerSelf(this.getBaseContext());
+
```
Comment (lazarkov) [inline examples/tv-app/android/App/platform-app/src/main/java/com/matter/tv/server/MainActivity.java:125]: Should be uncommented?

**Comment by xylophone21** [comment]
Hi @amitnj , sorry this PR is really big, could you please have some documents for how it works? Thanks

**Pull request closed without merging**